src/main/java/dao/impl/OperationsWithMapDbImpl.java

```diff
@@ -0,0 +1,34 @@
+package dao.impl;
+
+import dao.OperationsWithMapDb;
+import db.Storage;
+
+public class OperationsWithMapDbImpl implements OperationsWithMapDb {
+    private static final String EXCEPTION_MESSAGE = "There is no such fruit in the storage!";
+
+    @Override
+    public void writeToStorage(String fruit, Integer number) {
+        int newNumber = number;
+        if (Storage.storage.containsKey(fruit)) {
+            int previousNumber = Storage.storage.get(fruit);
+            newNumber = previousNumber + number;
+        }
+        Storage.storage.put(fruit, newNumber);
+    }
+
+    @Override
+    public boolean isInStorage(String fruit) {
+        return Storage.storage.containsKey(fruit);
+    }
+
+    @Override
+    public void subtrackFromStorage(String fruit, Integer subtractNumber) {
+        if (!Storage.storage.containsKey(fruit)) {
+            throw new RuntimeException(EXCEPTION_MESSAGE);
+        }
+        int previousNumber = Storage.storage.get(fruit);
+        int newNumber = previousNumber - subtractNumber;
+        Storage.storage.put(fruit, newNumber);
+
```

Comment:
The Data Access Object layer must not contain any business logic, like subtracting and addition and [there are some roles](https://www.baeldung.com/java-dao-pattern) how you can name your method for interaction with db entities.